File: apps/notification/src/functions/Authorizer.ts
'use strict'

import {Handler} from "aws-lambda";

const AbstractAuthorizer = require('../../../../shared/common/authorization/AbstractAuthorizer')
const {AuthPolicy, HttpVerb} = require('../../../../shared/common/authorization/AuthPolicy')

class Authorizer extends AbstractAuthorizer {
    generatePermissions(authPolicy: typeof AuthPolicy, userId: string) {
        if (userId) {
            authPolicy.allowMethod(HttpVerb.POST, `/v1/users/${userId}/groups`)
            authPolicy.allowMethod(HttpVerb.POST, `/v1/users/${userId}/groups/*/messages`)
            authPolicy.allowMethod(HttpVerb.GET, `/v1/users/${userId}/groups`)
            authPolicy.allowMethod(HttpVerb.GET, `/v1/users/${userId}/groups/*`)
        } else {
            authPolicy.denyAllMethods()
        }
    }
}

const authorizer = new Authorizer()

export const authorize: Handler = async (event) => {
    return await authorizer.authorize(event)
}

File: apps/authentication/src/functions/authorizer.py
from authorization.abstract_authorizer import AbstractAuthorizer
from authorization.auth_policy import AuthPolicy, HttpVerb


class Authorizer(AbstractAuthorizer):
    def __init__(self):
        AbstractAuthorizer.__init__(self)

    def generate_permissions(self, auth_policy: AuthPolicy, user_id: str):
        if user_id is not None:
            auth_policy.allow_method(HttpVerb.POST, f"/v1/users/{user_id}/sendCode")
            auth_policy.allow_method(HttpVerb.POST, f"/v1/users/{user_id}/confirmCode")
        else:
            auth_policy.deny_all_methods()


authorizer = Authorizer()


def authorize(event, context):
    return authorizer.authorize(event)
